paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/ProductResponse.java

```diff
@@ -0,0 +1,36 @@
+package com.chrisworks.paystackclient.domain.product;
+
+import com.chrisworks.paystackclient.domain.Amount;
+import com.chrisworks.paystackclient.domain.Currency;
+import com.chrisworks.paystackclient.domain.response.PaystackMultiResponse;
+import com.chrisworks.paystackclient.domain.response.PaystackSingleResponse;
+
+import java.math.BigInteger;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+public record ProductResponse(String name, String description, Currency currency, Amount.MoneyValue price,
+                              Integer quantity, boolean is_shippable, boolean unlimited, BigInteger integration,
+                              String domain, Metadata metadata, String slug, String product_code, Integer quantity_sold,
+                              String type, ShippingFields shipping_fields, boolean active, boolean in_stock,
+                              Integer minimum_orderable, Integer maximum_orderable, boolean low_stock_alert,
+                              BigInteger id, ZonedDateTime createdAt, ZonedDateTime updatedAt) {
+
+    public Amount getPrice() {
+        return this.price.ofCurrency(this.currency);
+    }
+
+    public record Single(boolean status, String message, ProductResponse data)
+            implements PaystackSingleResponse<ProductResponse> {
+    }
+
+    public record Multiple(boolean status, String message, List<ProductResponse> data, PageMetaInfo.Impl meta)
+            implements PaystackMultiResponse<ProductResponse> {
+    }
+
+    private record Metadata(String background_color) {
```

Comment:
Remove this class and substitute its usage with `Object` or `Map<String, Object>`

Reply:
Done